wgsl/index.bs

```diff
@@ -395,21 +411,103 @@ The following types are <dfn dfn noexport>host-shareable</dfn>:
 * [[#array-types]] if the array type has a stride attribute and its element type is host-shareable
 * [[#struct-types]] if all its members are host-shareable
 
+The contents of uniform buffers and storage buffers must be of host-shareable type.
+
 TODO: *This is a stub*: Collectively, the *stride* and *offset* attributes are called *layout attributes*.
 
 ### Storage Classes TODO ### {#storage-class}
 
+Memory locations are partitioned into <dfn noexport>storage classes</dfn>.
+Each storage class has unique properties affecting mutability, visibility, and the values they may contain.
+
+
+<table class='data'>
+  <thead>
+    <tr><td>Storage class
+        <td>Readable by shader?<br>Writable by shader?
+        <td>Sharing among invocations
+        <td>Scope
+        <td>Restrictions on stored values
+        <td>Notes
+  </thead>
+  <tr><td>input
+      <td>Read-only
+      <td>Same invocation only
+      <td>[=Module scope=]
+      <td>[=IO-shareable=]
+      <td>Contains values written by an upstream pipeline stage, or by the implementation.
+  <tr><td>output
+      <td>Read-write
+      <td>Same invocation only
+      <td>[=Module scope=]
+      <td>[=IO-shareable=]
+      <td>Shader writes values for downstream pipeline processing.
+  <tr><td>function
+      <td>Read-write
+      <td>Same invocation only
+      <td>[=Function scope=]
+      <td>[=Storable=]
+      <td>
+  <tr><td>private
+      <td>Read-write
+      <td>Same invocation only
+      <td>[=Module scope=]
+      <td>[=Storable=]
+      <td>
+  <tr><td>workgroup
+      <td>Read-write
+      <td>Invocations in the same [=workgroup=]
+      <td>[=Module scope=]
+      <td>[=Storable=]
+      <td>
+  <tr><td>uniform
+      <td>Read-only
+      <td>Invocations in the same [=shader stage=]
+      <td>[=Module scope=]
+      <td>[=Host-shareable=]
+      <td>See [[#memory-layout]]
+  <tr><td>storage_buffer
+      <td>Readable.<br>
+          Also writable if the variable is not read-only.
+      <td>Invocations in the same [=shader stage=]
+      <td>[=Module scope=]
+      <td>[=Host-shareable=]
+      <td>See [[#memory-layout]].
+  <tr><td>image
```

Comment:
Are we using "texture" and "image" interchangeably here? I understand that the term came from SPIR-V, where the relevant Vulkan API objects are images. In WGSL, it would be great to be consistent with WebGPU.

Reply:
I'll change to "texture"